### ISSUE TITLE
Exception in thread "DefaultShutdownHook" while running with JSystem maven plugin

### ISSUE THREAD

**Issue by NimrodTicotzner** [Top-Q/jsystem#227]

**Comment by igivon** [comment]
Maybe this will help:
it seems the issue is related to Maven version with version 2.2.1 its working, while with >3. it s not.

**Plugin Class Loader**
For each plugin used by a project, a plugin class loader is created as a child of the project class loader. If the project has no extensions and as such no project class loader, the plugin class loader will be a direct child of the Maven core class loader instead.
The class path of a plugin class loader is derived from the dependencies of the plugin as defined in the POM of the plugin artifact and optionally by user-defined plugin dependencies listed in the <plugin> declaration of the project that uses the plugin. The plugin class loader does neither contain the dependencies of the current project nor the compiled main/test classes of the project. Plugins that need to load classes from the compile/runtime/test class path of a project need to create a custom URLClassLoader in combination with the mojo annotation @requiresDependencyResolution.
To sum up, a plugin can access classes/resources from the Java bootstrap class path, from the dependencies of the plugin, from the Maven API and from the system class loader. To load other classes, a custom class loader needs to be created.

Maven APi changed and Ant run is effected: [](https://cwiki.apache.org/confluence/display/MAVEN/Maven+3.x+Plugin+Compatibility+Matrix)

**Comment by itaiag** [comment]
This seems to be happening in Maven 3. Probably not happening in 2.2.1
